### PR TITLE
Rewrite `ScoreV1/osu!` using ground truth information

### LINKED ISSUE #2528: osu! scoring info is wrong
https://osu.ppy.sh/wiki/en/Gameplay/Score/osu%21

looks like the original author of this made a lot of assumptions that just aren't true. spinner score bonuses are wrong & the normal spin points are not mentioned anywhere, difficulty multipliers do not affect the score in the way stated, there's some random mixed-up info about scorev2 here and there (#2026, #2025), some of the formulas appear more complicated/confusing than they really are (https://github.com/ppy/osu-wiki/pull/2470#discussion_r322029818)

and I didn't check yet but I'd assume some of these problems apply to the other game modes too

### PR DESCRIPTION
**Pull request by bdach**
- Closes #2528
- Step forward in #2026 (1/4)

The contents were not as wrong as the original issue described them, because things improved along the way (e.g. in e5e1f8f8909d6037c207a8ac906898e59b2d0a4b). That said, there were still some half-truths and inaccuracies in the old version.

I've opted for a full rewrite using a bit of a different structure to make describing the algorithm easier. This is going to get much worse when it comes to taiko, for instance, as there's a lot of object type specific logic there. Interested in opinions about how well this style holds up.

Information was cross-checked against source and also verified empirically to a degree. If there's anything unclear in the wording I used please let me know - describing complex formulas and algorithms tends to get messy and imprecise when done in natural language.

## Self-check

- [x] The changes are tested against the [contribution checklist](https://osu.ppy.sh/wiki/osu!_wiki/Contribution_guide#self-check)

### CODE REVIEW

File: wiki/Gameplay/Score/ScoreV1/osu!/en.md [hunk 1]
```diff
@@ -1,31 +1,51 @@
...
+## Hit circles
+
+Each hit circle is assigned a numerical point value using the formula below:
+
+`Score = Hit value * ((Combo multiplier * Difficulty multiplier * Mod multiplier) / 25)`
+
+where:
+
+- The *hit value* is the numerical value of the hit circle judgement (50, 100, or 300).
+- The *combo multiplier* is equal to (combo before this hit - 1) or 0, whichever is higher.
```
Comment (Walavouchey) [inline wiki/Gameplay/Score/ScoreV1/osu!/en.md:19]: so with a combo of 0 or 1 (start of beatmap or after each combo reset) you get 0 points for the next object?

File: wiki/Gameplay/Score/ScoreV1/osu!/en.md [hunk 2]
```diff
@@ -1,31 +1,51 @@
...
+
+The **difficulty multiplier** is equal to an older version of star rating for the beatmap being played. It can be calculated via the following formula:
+
+`Difficulty multiplier = Round((HP Drain + Circle Size + Overall Difficulty + Clamp(Hit object count / Drain time in seconds * 8, 0, 16)) / 38 * 5)`
+
+Note that game modifiers (like Hard Rock or Easy, which change circle size, for instance) do not affect the difficulty multiplier, as the original values of the variables are always used in the formula above regardless of which mods are enabled.
+
+## Sliders
+
+Each slider as a whole produces a 50, 100, or 300 judgement, based on the proportion of slider ticks hit. This judgement is converted to a score value using the same method that hit circles use.
```
Comment (Walavouchey) [inline wiki/Gameplay/Score/ScoreV1/osu!/en.md:33]: > based on the proportion of slider ticks hit

not including the slider head, end and repeats?

Comment (bdach) [reply wiki/Gameplay/Score/ScoreV1/osu!/en.md:33]: Including, but not sure how to word it nicely. The lazer term we use for this is "nested objects". Not sure it makes sense for the wiki.

File: wiki/Gameplay/Score/ScoreV1/osu!/en.md (hunk 1 above)
Comment (bdach) [reply wiki/Gameplay/Score/ScoreV1/osu!/en.md:19]: No, this is just wrong. Clerical error when touching the old (correct) formula. Fixed in https://github.com/ppy/osu-wiki/pull/7525/commits/d2f28f879d4a17ee154bf25ca3c586d6f38a8621.

File: wiki/Gameplay/Score/ScoreV1/osu!/en.md (hunk 2 above)
Comment (bdach) [reply wiki/Gameplay/Score/ScoreV1/osu!/en.md]: Have attempted to clarify in https://github.com/ppy/osu-wiki/pull/7525/commits/3851987283121782c630f3947787e393ecbe9391, please check if it helps. Suggestions for alternatives welcome.

File: wiki/Gameplay/Score/ScoreV1/osu!/en.md [hunk 3]
```diff
@@ -1,31 +1,51 @@
...
+
+The **difficulty multiplier** is equal to an older version of star rating for the beatmap being played. It can be calculated via the following formula:
+
+`Difficulty multiplier = Round((HP Drain + Circle Size + Overall Difficulty + Clamp(Hit object count / Drain time in seconds * 8, 0, 16)) / 38 * 5)`
+
+Note that game modifiers (like Hard Rock or Easy, which change circle size, for instance) do not affect the difficulty multiplier, as the original values of the variables are always used in the formula above regardless of which mods are enabled.
+
+## Sliders
+
+Each slider as a whole produces a 50, 100, or 300 judgement, based on the proportion of slider parts (namely, the slider head, slider end, slider ticks, and slider repeats) hit. This judgement is converted to a score value using the same method that hit circles use.
```
Comment (Walavouchey) [inline wiki/Gameplay/Score/ScoreV1/osu!/en.md:33]: may be worth clarifying what the proportions are, same for spinners

Comment (bdach) [reply wiki/Gameplay/Score/ScoreV1/osu!/en.md:33]: Reason why I haven't done this is that I wasn't sure if that belongs in here or in the judgement article. I'm fine with including here if you are.

Comment (Walavouchey) [reply wiki/Gameplay/Score/ScoreV1/osu!/en.md:33]: ah, you're right

Comment (bdach) [reply wiki/Gameplay/Score/ScoreV1/osu!/en.md]: Addressed in https://github.com/ppy/osu-wiki/pull/7527

**Review by TicClick — APPROVED** [review]

File: wiki/Gameplay/Score/ScoreV1/osu!/en.md [hunk 4]
```diff
@@ -1,31 +1,50 @@
...
+
+Each hit circle is assigned a numerical point value using the formula below:
+
+`Score = Hit value * (1 + (Combo multiplier * Difficulty multiplier * Mod multiplier / 25))`
+
+where:
+
+- The *hit value* is the numerical value of the hit circle judgement (50, 100, or 300).
+- The *combo multiplier* is equal to (combo before this hit - 1) or 0, whichever is higher.
+- The *difficulty multiplier* is a value specific to the beatmap being played. The exact formula to compute it is given in the next subsection.
```
Comment (Walavouchey) [inline wiki/Gameplay/Score/ScoreV1/osu!/en.md:19]:
```suggestion
- The *difficulty multiplier* is a value specific to the beatmap being played. See the [below section](#difficulty-multiplier) for the exact formula to compute it.
```

File: wiki/Gameplay/Score/ScoreV1/osu!/en.md [hunk 5]
```diff
@@ -1,31 +1,50 @@
...
+`Difficulty multiplier = Round((HP Drain + Circle Size + Overall Difficulty + Clamp(Hit object count / Drain time in seconds * 8, 0, 16)) / 38 * 5)`
+
+Note that game modifiers (like Hard Rock or Easy, which change circle size, for instance) do not affect the difficulty multiplier, as the original values of the variables are always used in the formula above regardless of which mods are enabled.
+
+## Sliders
+
+Each slider as a whole produces a 50, 100, or 300 judgement, based on the proportion of slider parts (namely, the slider head, slider tail, slider ticks, and slider repeats) hit. This judgement is converted to a score value using the same method that hit circles use.
 
-`Score = Hit Value + (Hit Value * ((Combo multiplier * Difficulty multiplier * Mod multiplier) / 25))`
+Additionally, elements of the slider grant score in an independent fashion:
```
Comment (Walavouchey) [inline wiki/Gameplay/Score/ScoreV1/osu!/en.md:34]:
```suggestion
Additionally, elements of the slider grant score in an independent fashion, unaffected by any bonuses or multipliers:
```

File: wiki/Gameplay/Score/ScoreV1/osu!/en.md [hunk 6]
```diff
@@ -1,31 +1,50 @@
...
-| :-: | :-- |
-| **Hit Value** | The hit circle judgement (50, 100 or 300), any slider ticks and spinner's bonus |
-| **Combo multiplier** | (Combo before this hit - 1) or 0; whichever is higher |
-| **Difficulty multiplier** | The difficulty setting for the beatmap (see next header) |
-| **Mod multiplier** | The multiplier of the selected mods |
+- Each slider tick hit grants 10 points.
+- Each slider repeat or slider tail hit grants 30 points.
 
-Additionally each slider start, end and repeat tick awards 30 points, each slider middle tick awards 10 points and each spin of a spinner awards 100 points.
+The point values above are not affected by any bonuses or multipliers.
```
Comment (Walavouchey) [inline wiki/Gameplay/Score/ScoreV1/osu!/en.md:40]:
```suggestion
```

File: wiki/Gameplay/Score/ScoreV1/osu!/en.md [hunk 7]
```diff
@@ -1,31 +1,50 @@
...
 
-Additional bonus of 1,000 points given for each spin of a spinner after the spinner meter is full.
+## Spinners
 
-## Difficulty multiplier
+Each spinner as a whole produces a 50, 100, or 300 judgement, based on the ratio of rotations performed to rotations required to complete the spinner. This judgement is converted to a score value using the same method that hit circles use.
 
-The **Difficulty multiplier** equals the old star rating. It can be calculated via the following formula:
+Additionally, the spinner grants bonus points:
```
Comment (Walavouchey) [inline wiki/Gameplay/Score/ScoreV1/osu!/en.md:45]:
```suggestion
The spinner also grants additional bonus points, unaffected by any other bonuses or multipliers:
```

File: wiki/Gameplay/Score/ScoreV1/osu!/en.md [hunk 8]
```diff
@@ -1,31 +1,50 @@
...
 
-`Stars = Round((HP Drain + Circle Size + Overall Difficulty + Clamp(Hit object count / Drain time in seconds * 8, 0, 16)) / 38 * 5)`
+- Each full spin before the spinner is completed grants 100 points.
+- After the spinner is completed, full spins grant an additional 1,000 points, for a total of 1,100 points altogether per spin.
```
Comment (Walavouchey) [inline wiki/Gameplay/Score/ScoreV1/osu!/en.md:49]:
```suggestion
```

File: wiki/Gameplay/Score/ScoreV1/osu!/en.md [hunk 9]
```diff
@@ -1,31 +1,50 @@
...
 
-Note that game modifiers (like Hard Rock/Easy) will not change the **Difficulty multiplier**.
-It will only account for original values only.
+The point values above are not affected by any bonuses or multipliers.
```
Comment (Walavouchey) [inline wiki/Gameplay/Score/ScoreV1/osu!/en.md:50]:
```suggestion
```

File: wiki/Gameplay/Score/ScoreV1/osu!/en.md [hunk 10]
```diff
@@ -1,31 +1,50 @@
...
+
+The **difficulty multiplier** is equal to an older version of star rating for the beatmap being played. It can be calculated via the following formula:
+
+`Difficulty multiplier = Round((HP Drain + Circle Size + Overall Difficulty + Clamp(Hit object count / Drain time in seconds * 8, 0, 16)) / 38 * 5)`
+
+Note that game modifiers (like Hard Rock or Easy, which change circle size, for instance) do not affect the difficulty multiplier, as the original values of the variables are always used in the formula above regardless of which mods are enabled.
+
+## Sliders
+
+Each slider as a whole produces a 50, 100, or 300 judgement, based on the proportion of slider parts (namely, the slider head, slider tail, slider ticks, and slider repeats) hit. This judgement is converted to a score value using the same method that hit circles use.
```
Comment (Walavouchey) [inline wiki/Gameplay/Score/ScoreV1/osu!/en.md:32]:
```suggestion
Each slider as a whole produces a 50, 100, or 300 judgement, based on the proportion of slider parts hit (namely, the slider head, slider tail, slider ticks, and slider repeats). This judgement is converted to a score value using the same method that hit circles use.
```

**Review by Walavouchey — APPROVED** [review]

**Pull request merged**